examples/src/test/scala/JedisExample.scala

```diff
@@ -58,83 +60,129 @@ object DataSources {
       _ <- fetchNumber(n)
       u <- HttpExample.fetchUser(n)
     } yield u
+
+  def fetchMulti[F[_]: ConcurrentEffect]: Fetch[F, List[HttpExample.User]] =
+    List(4, 5, 6).traverse(HttpExample.fetchUser[F](_))
 }
 
 object Binary {
+  type ByteArray = Array[Byte]
+
+  def byteOutputStream[F[_]](implicit S: Sync[F]): Resource[F, ByteArrayOutputStream] =
+    Resource.fromAutoCloseable(S.delay(new ByteArrayOutputStream()))
+
+  def byteInputStream[F[_]](bin: ByteArray)(
+      implicit S: Sync[F]): Resource[F, ByteArrayInputStream] =
+    Resource.fromAutoCloseable(S.delay(new ByteArrayInputStream(bin)))
+
+  def outputStream[F[_]](b: ByteArrayOutputStream)(
+      implicit S: Sync[F]): Resource[F, ObjectOutputStream] =
+    Resource.fromAutoCloseable(S.delay(new ObjectOutputStream(b)))
+
+  def inputStream[F[_]](b: ByteArrayInputStream)(
+      implicit S: Sync[F]): Resource[F, ObjectInputStream] =
+    Resource.fromAutoCloseable(S.delay(new ObjectInputStream(b)))
+
   def fromString(s: String): Array[Byte] =
     s.getBytes(Charset.forName("UTF-8"))
 
-  def serialize[F[_]: Sync, A](v: A): F[Array[Byte]] = {
-    val baos = new ByteArrayOutputStream()
-    val oos  = new ObjectOutputStream(baos)
-    Sync[F].bracket(Sync[F].pure(oos))({ in =>
-      Sync[F].pure({
-        in.writeObject(v)
-        in.flush()
-        baos.toByteArray
-      })
-    })({ in =>
-      Sync[F].delay(in.close())
+  def serialize[F[_], A](obj: A)(
+      implicit S: Sync[F]
+  ): F[ByteArray] = {
+    type Resources = (ObjectOutputStream, ByteArrayOutputStream)
+
+    def resources[F[_]](implicit S: Sync[F]): Resource[F, Resources] =
+      for {
+        byte <- byteOutputStream
```

Comment:
We can extract a `flatProduct` function, of the type `F[A] => (A => F[B]) => F[(A, B)]`. Maybe there is something like that already.

Reply:
Found it: https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/FlatMap.scala#L114 Good catch!